cmd/libs/go2idl/make-client/generators/client-generator.go

```diff
@@ -0,0 +1,270 @@
+	}
+}
+
+// DefaultNameSystem returns the default name system for ordering the types to be
+// processed by the generators in this package.
+func DefaultNameSystem() string {
+	return "public"
+}
+
+// Packages makes the sets package definition.
+func Packages(context *generator.Context, arguments *args.GeneratorArgs) generator.Packages {
+	boilerplate, err := arguments.LoadGoBoilerplate()
+	if err != nil {
+		glog.Fatalf("Failed loading boilerplate: %v", err)
+	}
+
+	// @lavalamp: This is quite hacky. Any suggestions?
+	topLevelTypes := map[*types.Type]struct{}{}
+	ref := context.Universe.Get(types.Name{"k8s.io/kubernetes/cmd/libs/go2idl/make-client/input", "ReferenceTypes"})
+	for _, m := range ref.Members {
+		topLevelTypes[m.Type] = struct{}{}
+	}
+	return generator.Packages{&generator.DefaultPackage{
+		PackageName: "unversionedClient",
+		PackagePath: arguments.OutputPackagePath,
+		HeaderText: append(boilerplate, []byte(
+			`
+// This file was autogenerated by the command:
+// $ `+strings.Join(os.Args, " ")+`
+// Do not edit it manually!
+
+`)...),
+		PackageDocumentation: []byte(
+			`// Package unversionedClient has the automatically generated clients for unversioned resources.
+`),
+		// GeneratorFunc returns a list of generators. Each generator makes a
+		// single file.
+		GeneratorFunc: func(c *generator.Context) (generators []generator.Generator) {
+			generators = []generator.Generator{
+				// Always generate a "doc.go" file.
+				generator.DefaultGen{OptionalName: "doc"},
+			}
+			// Since we want a file per type that we generate a set for, we
+			// have to provide a function for this.
+			for _, t := range c.Order {
+				generators = append(generators, &genClient{
+					DefaultGen: generator.DefaultGen{
+						// Use the privatized version of the
+						// type name as the file name.
+						//
+						// TODO: make a namer that converts
+						// camelCase to '-' separation for file
+						// names?
+						OptionalName: c.Namers["private"].Name(t),
+					},
+					outputPackage: arguments.OutputPackagePath,
+					typeToMatch:   t,
+					imports:       generator.NewImportTracker(),
+				})
+			}
+			return generators
+		},
+		FilterFunc: func(c *generator.Context, t *types.Type) bool {
+			_, found := topLevelTypes[t]
+			return found
+		},
+	}}
+}
+
+// genClient produces a file with a set for a single type.
+type genClient struct {
+	generator.DefaultGen
+	outputPackage string
+	typeToMatch   *types.Type
+	imports       *generator.ImportTracker
+}
+
+// Filter ignores all but one type because we're making a single file per type.
+func (g *genClient) Filter(c *generator.Context, t *types.Type) bool { return t == g.typeToMatch }
+
+func (g *genClient) Namers(c *generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer(g.outputPackage, g.imports),
+	}
+}
+
+func (g *genClient) Imports(c *generator.Context) (imports []string) {
+	return append(g.imports.ImportLines())
```

Comment:
nit: if you don't need to add any "additional" imports, just return g.imports.ImportLines()